src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs

```diff
@@ -0,0 +1,353 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace CreateBaselineUpdatePR;
+
+using Octokit;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+public class PRCreator
+{
+    private readonly string _repoOwner;
+    private readonly string _repoName;
+    private readonly GitHubClient _client;
+    private const string BuildLink = "https://dev.azure.com/dnceng/internal/_build/results?buildId=";
+    private const string TreeMode = "040000";
+
+    public PRCreator(string repo, string gitHubToken)
+    {
+        // Create a new GitHub client
+        _client = new GitHubClient(new ProductHeaderValue(System.Reflection.Assembly.GetExecutingAssembly().GetName().Name));
+        var authToken = new Credentials(gitHubToken);
+        _client.Credentials = authToken;
+        _repoOwner = repo.Split('/')[0];
+        _repoName = repo.Split('/')[1];
+    }
+
+    private static readonly string DefaultLicenseBaselineContent = "{\n  \"files\": []\n}";
```

Comment:
Why static readonly vs a const?  Regardless this feels better places next to the consts

Reply:
Whoops. Forgot to update this. Will fix.